tests/test_types.py

```diff
@@ -2774,3 +2776,147 @@ class Model(BaseModel):
         {'loc': ('my_none_dict', 'a'), 'msg': 'value is not None', 'type': 'type_error.not_none'},
         {'loc': ('my_json_none',), 'msg': 'value is not None', 'type': 'type_error.not_none'},
     ]
+
+
+@pytest.mark.parametrize(
+    'value,result',
+    (
+        ('1996-01-22', date(1996, 1, 22)),
+        (date(1996, 1, 22), date(1996, 1, 22)),
+    ),
+)
+def test_past_date_validation_success(value, result):
+    class Model(BaseModel):
+        foo: PastDate
+
+    assert Model(foo=value).foo == result
+
+
+@pytest.mark.parametrize(
+    'value,errors',
+    (
+        (
+            date.today(),
+            [
+                {
+                    'loc': ('foo',),
+                    'msg': f'date "{date.today()}" is not in the past',
+                    'type': 'value_error.date.not_in_the_past',
+                    'ctx': {'date': str(date.today())},
+                }
+            ],
+        ),
+        (
+            date.today() + timedelta(1),
+            [
+                {
+                    'loc': ('foo',),
+                    'msg': f'date "{date.today() + timedelta(1)}" is not in the past',
+                    'type': 'value_error.date.not_in_the_past',
+                    'ctx': {'date': str(date.today() + timedelta(1))},
+                }
+            ],
+        ),
+        (
+            datetime.today(),
+            [
+                {
+                    'loc': ('foo',),
+                    'msg': f'date "{date.today()}" is not in the past',
+                    'type': 'value_error.date.not_in_the_past',
+                    'ctx': {'date': str(date.today())},
+                }
+            ],
+        ),
+        (
+            datetime.today() + timedelta(1),
+            [
+                {
+                    'loc': ('foo',),
+                    'msg': f'date "{date.today() + timedelta(1)}" is not in the past',
+                    'type': 'value_error.date.not_in_the_past',
+                    'ctx': {'date': str(date.today() + timedelta(1))},
+                }
+            ],
+        ),
+    ),
+)
+def test_past_date_validation_fails(value, errors):
```

Comment:
You could probably just write an input value and parse value as params and write the whole error object in the test

Reply:
True. Going to apply this tonight, thanks :)

Reply:
I've applied some modifications. Is that what you meant? @PrettyWood

Reply:
Yes thanks!